group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -1248,29 +1248,118 @@ private CompletableFuture<Map<String, Errors>> persisterDeleteToGroupIdErrorMap(
+                        deleteShareGroupStateRequestTopicsData.add(new DeleteShareGroupStateRequestData.DeleteStateData()
+                            .setTopicId(topicId)
+                            .setPartitions(partitions));
+                    }
+                } else {
+                    deleteShareGroupOffsetsResponseTopicList.add(new DeleteShareGroupOffsetsResponseData.DeleteShareGroupOffsetsResponseTopic()
+                        .setTopicName(topic.topicName())
+                        .setPartitions(topic.partitions().stream().map(
+                            partition -> new DeleteShareGroupOffsetsResponseData.DeleteShareGroupOffsetsResponsePartition()
+                                .setPartitionIndex(partition)
+                                .setErrorCode(Errors.UNKNOWN_TOPIC_OR_PARTITION.code())
+                                .setErrorMessage(Errors.UNKNOWN_TOPIC_OR_PARTITION.message())
+                        ).toList()));
+                }
+            });
+
+            // If the request for the persister is empty, just complete the operation right away.
+            if (deleteShareGroupStateRequestTopicsData.isEmpty()) {
+                return CompletableFuture.completedFuture(new DeleteShareGroupOffsetsResponseData().setResponses(deleteShareGroupOffsetsResponseTopicList));
+            }
+
+            return sendPersisterDeleteStateRequest(
+                requestData,
+                requestTopicIdToNameMapping,
+                deleteShareGroupStateRequestTopicsData,
+                deleteShareGroupOffsetsResponseTopicList
+            );
+        }).exceptionally(throwable -> {
+            log.error("Failed to get initialized topic partitions for the group {}", groupId, throwable);
+            return DeleteShareGroupOffsetsRequest.getErrorDeleteResponseData(Errors.forException(throwable));
+        });
+    }
+
+    private CompletableFuture<DeleteShareGroupOffsetsResponseData> checkIfGroupIsEmptyAndProcess(
+        String groupId,
+        DeleteShareGroupOffsetsRequestData requestData
+    ) {
+        // This is done to make sure the provided group is empty. Offsets can be deleted only for an empty share group.
+        CompletableFuture<List<ShareGroupDescribeResponseData.DescribedGroup>> describeGroupFuture =
+            runtime.scheduleReadOperation(
+                "share-group-describe",
+                topicPartitionFor(groupId),
+                (coordinator, lastCommittedOffset) -> coordinator.shareGroupDescribe(List.of(groupId), lastCommittedOffset)
+            ).exceptionally(exception -> handleOperationException(
+                "share-group-describe",
+                List.of(groupId),
+                exception,
+                (error, __) -> ShareGroupDescribeRequest.getErrorDescribedGroupList(List.of(groupId), error),
+                log
+            ));
+
+        return describeGroupFuture.thenCompose(groups -> {
+            if (groups == null || groups.isEmpty()) {
```

Comment:
I would be better if you have a separate method to encapsulate all this logic
will help with testing and readability.